gui/cloudconnectionwizard.cpp

```diff
@@ -0,0 +1,669 @@
+	_label3->setEnabled(false);
+}
+
+void CloudConnectionWizard::hideStepManualModeFailure() {
+	hideContainer();
+	hideBackButton();
+	removeWidgetChecked(_label0);
+	removeWidgetChecked(_label1);
+	removeWidgetChecked(_label2);
+	removeWidgetChecked(_label3);
+}
+
+void CloudConnectionWizard::showStepManualModeSuccess() {
+	_headlineLabel->setLabel(_("Manual Mode: Success"));
+	showContainer("ConnectionWizard_Success");
+	_closeButton->setVisible(false);
+
+	_label0 = new StaticTextWidget(_container, "ConnectionWizard_Success.Line1", _("Your cloud storage has been connected!"));
+	_button0 = new ButtonWidget(this, "GlobalOptions_Cloud_ConnectionWizard.FinishButton", _("Finish"), Common::U32String(), kCloseCmd);
+}
+
+void CloudConnectionWizard::hideStepManualModeSuccess() {
+	hideContainer();
+	_closeButton->setVisible(true);
+	removeWidgetChecked(_label0);
+	removeWidgetChecked(_button0);
+}
+
+// utils
+
+void CloudConnectionWizard::showContainer(const Common::String &dialogName) {
+	_container = new ScrollContainerWidget(this, "GlobalOptions_Cloud_ConnectionWizard.Container", dialogName, kCloudConnectionWizardReflowCmd);
+	_container->setTarget(this);
+	_container->setBackgroundType(ThemeEngine::kWidgetBackgroundNo);
+}
+
+void CloudConnectionWizard::hideContainer() {
+	removeWidgetChecked(_container);
+}
+
+void CloudConnectionWizard::showBackButton() {
+	_prevStepButton = new ButtonWidget(this, "GlobalOptions_Cloud_ConnectionWizard.PrevButton", _("Back"), Common::U32String(), kCloudConnectionWizardBackButtonCmd);
+}
+
+void CloudConnectionWizard::hideBackButton() {
+	removeWidgetChecked(_prevStepButton);
+}
+
+void CloudConnectionWizard::showNextButton() {
+	_nextStepButton = new ButtonWidget(this, "GlobalOptions_Cloud_ConnectionWizard.NextButton", _("Next"), Common::U32String(), kCloudConnectionWizardNextButtonCmd);
+}
+
+void CloudConnectionWizard::hideNextButton() {
+	removeWidgetChecked(_nextStepButton);
+}
+
+void CloudConnectionWizard::removeWidgetChecked(ScrollContainerWidget *&widget) {
+	if (widget) {
+		removeWidget(widget);
+		widget = nullptr;
+	}
+}
+
+void CloudConnectionWizard::removeWidgetChecked(ButtonWidget *&widget) {
+	if (widget) {
+		removeWidget(widget);
+		widget = nullptr;
+	}
+}
+
+void CloudConnectionWizard::removeWidgetChecked(StaticTextWidget *&widget) {
+	if (widget) {
+		removeWidget(widget);
+		widget = nullptr;
+	}
+}
+
+void CloudConnectionWizard::removeWidgetChecked(EditTextWidget *&widget) {
+	if (widget) {
+		removeWidget(widget);
+		widget = nullptr;
+	}
+}
+
+// logic
+
+void CloudConnectionWizard::storageConnectionCallback(Networking::ErrorResponse response) {
+	if (response.failed || response.interrupted) {
+		return;
+	}
+
+	_switchToSuccess = true;
+}
+
+void CloudConnectionWizard::manualModeConnect() {
+	if (_connecting)
+		return;
+
+	if (_label1)
+		_label1->setLabel(Common::U32String());
+
+	// get the code entered
+	Common::String code = "";
```

Comment:
This assignment is redundant: the String constructor initializes it with an empty string already.